ctk/idp/src/main/kotlin/org/codice/compliance/web/sso/error/PostSSOErrorTest.kt

```diff
@@ -16,129 +16,129 @@ package org.codice.compliance.web.sso.error
-        "POST AuthnRequest With Relay State Greater Than 80 Bytes Test" {
+        "Bindings 3.5.3: POST AuthnRequest With Relay State Greater Than 80 Bytes Test" {
             try {
                 Log.debugWithSupplier {
-                    "POST AuthnRequest With Relay State Greater Than 80 Bytes Test"
+                    "Bindings 3.5.3: POST AuthnRequest With Relay State Greater Than 80 Bytes Test"
                 }
                 val authnRequest = createDefaultAuthnRequest(SamlProtocol.Binding.HTTP_POST)
                 val encodedRequest =
                         signAndEncodeToString(authnRequest, RELAY_STATE_GREATER_THAN_80_BYTES)
                 val response = sendPostAuthnRequest(encodedRequest)
 
-                val samlResponseDom = response.getBindingVerifier().decodeAndVerifyError()
+                if (!isLenient || !BindingVerifier.isErrorHttpStatusCode(response.statusCode)) {
+                    val samlResponseDom = response.getBindingVerifier().decodeAndVerifyError()
 
-                CoreVerifier.verifyErrorStatusCode(samlResponseDom,
+                    CoreVerifier.verifyErrorStatusCode(samlResponseDom,
                         samlErrorCode = SAMLBindings_3_5_3_a,
                         expectedStatusCode = REQUESTER)
+                }
             } catch (e: SAMLComplianceException) {
                 throw SAMLComplianceException.recreateExceptionWithErrorMessage(e)
             }
         }
 
-        "Empty POST AuthnRequest Test" {
+        "Profiles 4.1.4.1: Empty POST AuthnRequest Test" {
             try {
-                Log.debugWithSupplier { "Empty POST AuthnRequest Test" }
+                Log.debugWithSupplier { "Profiles 4.1.4.1: Empty POST AuthnRequest Test" }
                 val authnRequest = AuthnRequestBuilder().buildObject().apply {
                 }
 
                 val encodedRequest = signAndEncodeToString(authnRequest, EXAMPLE_RELAY_STATE)
                 val response = sendPostAuthnRequest(encodedRequest)
 
-                val samlResponseDom = response.getBindingVerifier().decodeAndVerifyError()
+                if (!isLenient || !BindingVerifier.isErrorHttpStatusCode(response.statusCode)) {
+                    val samlResponseDom = response.getBindingVerifier().decodeAndVerifyError()
 
-                CoreVerifier.verifyErrorStatusCode(samlResponseDom,
+                    CoreVerifier.verifyErrorStatusCode(samlResponseDom,
                         samlErrorCode = SAMLProfiles_4_1_4_1_a,
                         expectedStatusCode = REQUESTER)
-                ProfilesVerifier(samlResponseDom).verifyErrorResponseAssertion()
+                    ProfilesVerifier(samlResponseDom).verifyErrorResponseAssertion()
+                }
             } catch (e: SAMLComplianceException) {
                 throw SAMLComplianceException.recreateExceptionWithErrorMessage(e)
             }
         }
 
-        "POST AuthnRequest With Empty Subject Test" {
```

Comment:
❓ Why was the `Empty Subject Test` removed/changed?

Reply:
The spec says
```
Note that the service provider MAY include a <Subject> element in the request that 
names the actual identity about which it wishes to receive an assertion. This element 
MUST NOT contain any <SubjectConfirmation> elements. If the identity provider does 
not recognize the principal as that identity, then it MUST respond with a <Response> 
message containing an error status and no assertions.
```
Because of that I changed it to send a principal the IdP doesn't recognize instead of an empty subject